### PR TITLE
Add tag filter to analytics specs

### PR DESCRIPTION
**Pull request by szekelyzol**
Changes are for [this Asana task](https://app.asana.com/0/1205634133195403/1207529250404620).

**Summary**

- Added the `tag` parameter value to the `filterBy` parameter for all Analytics endpoints
- Restructured the `filterBy` parameter into a parameter schema for reusability
- Fixed the way request query parameter values are described for `filterBy` by adding them as `style: deepObject` object elements.
- Fixed some example values
- Fixed the field format of `interval` and added its enums

### CODE REVIEW

**Comment by szekelyzol** [comment]
Hi @olivierapivideo, @ThibaultBee! I added you to this PR as reviewers because I am not sure if the OpenAPI client generator will like this solution or not. We already use `deepObject` as a parameter style for an existing endpoint (for adding video metadata), however, in that case we're using simple non-nested objects. The way `filterBy` is serialized **might be** undefined behavior for the OpenAPI specification. See specs `#Query-parameters` [here](https://swagger.io/docs/specification/serialization/). ⚠️ 

I tested this change with both the old and the new Swagger Editor, Doctave's validator, and the results were good ✅ 

Please help me validate that it would pass the client generator as well 🙏 Thanks!

File: openapi.yaml [hunk 1]
```diff
@@ -11964,27 +11943,7 @@ paths:
...
-          - `country`: Returns analytics based on the viewers' country. The list of supported country names are based on the [GeoNames public database](https://www.geonames.org/countries/).
-          - `deviceType`: Returns analytics based on the type of device used by the viewers. Possible response values are: `computer`, `phone`, `tablet`, `tv`, `console`, `wearable`, `unknown`.
-          - `operatingSystem`: Returns analytics based on the operating system used by the viewers. Response values include `windows`, `mac osx`, `android`, `ios`, `linux`.
-          - `browser`: Returns analytics based on the browser used by the viewers. Response values include `chrome`, `firefox`, `edge`, `opera`.
-        style: form
-        explode: true
-        schema:
-          type: string
-        example: filterBy[continent]=EU&filterBy[country]=FR&filterBy[browser]=Safari&filterBy[browser]=Firefox
+      - $ref: '#/components/parameters/filterBy'
```
Comment (olivierapivideo) [inline openapi.yaml:11946]: I think we are missing the pagination details here:
```
      - $ref: '#/components/parameters/current-page'
      - $ref: '#/components/parameters/page-size'
```

File: openapi.yaml [hunk 2]
```diff
@@ -12192,29 +12151,9 @@ paths:
...
-          - `operatingSystem`: Returns analytics based on the operating system used by the viewers. Response values include `windows`, `mac osx`, `android`, `ios`, `linux`.
-          - `browser`: Returns analytics based on the browser used by the viewers. Response values include `chrome`, `firefox`, `edge`, `opera`.
-        style: form
-        explode: true
-        schema:
-          type: string
-        example: filterBy[continent]=EU&filterBy[country]=FR&filterBy[browser]=Safari&filterBy[browser]=Firefox
+          enum: [hour, day]
+        example: hour
+      - $ref: '#/components/parameters/filterBy'
```
Comment (olivierapivideo) [inline openapi.yaml:12156]: I think we are missing the pagination details here:
```
      - $ref: '#/components/parameters/current-page'
      - $ref: '#/components/parameters/page-size'
```

Comment (mioara-api-video) [reply openapi.yaml:12156]: Yes, but beware, they're camelCase: `currentPage` and `pageSize`

File: openapi.yaml [hunk 3]
```diff
@@ -15104,6 +15043,65 @@ components:
         metadata:
           $ref: '#/components/schemas/video-status-encoding-metadata'
   parameters:
+    filterBy:
+      name: filterBy
+      description: |
+          Use this parameter to filter the API's response based on different data dimensions. You can serialize filters in your query to receive more detailed breakdowns of your analytics.
+
+          - If you do not set a value for `filterBy`, the API returns the full dataset for your project.
+          - The API only accepts the `mediaId` and `mediaType` filters when you call `/data/metrics/play/total`.
```
Comment (mioara-api-video) [inline openapi.yaml:15052]: And `data/buckets/play-total/media-id`

File: openapi.yaml [hunk 4]
```diff
@@ -15104,6 +15043,65 @@ components:
...
+          
+          These are the available breakdown dimensions:
+          
+          - `mediaId`: Returns analytics based on the unique identifiers of a video or a live stream.
+          - `mediaType`: Returns analytics based on the type of content. Possible values: `video` and `live-stream`. 
+          - `continent`: Returns analytics based on the viewers' continent. The list of supported continents names are based on the [GeoNames public database](https://www.geonames.org/countries/). You must use the ISO-3166 alpha2 format, for example `EU`. Possible values are: `AS`, `AF`, `NA`, `SA`, `AN`, `EU`, `AZ`. 
+          - `country`: Returns analytics based on the viewers' country. The list of supported country names are based on the [GeoNames public database](https://www.geonames.org/countries/). You must use the ISO-3166 alpha2 format, for example `FR`.
+          - `deviceType`: Returns analytics based on the type of device used by the viewers. Possible response values are: `computer`, `phone`, `tablet`, `tv`, `console`, `wearable`, `unknown`.
```
Comment (mioara-api-video) [inline openapi.yaml:15060]: Shouldn't we write the second sentence as "**Examples of** possible response values are", to show that list is non-exhaustive? I'm fairly certain I've seen "Desktop" for instance

File: openapi.yaml [hunk 5]
```diff
@@ -15104,6 +15043,65 @@ components:
...
+            description: Returns analytics based on the viewers' continent. The list of supported continents names are based on the [GeoNames public database](https://www.geonames.org/countries/). You must use the ISO-3166 alpha2 format, for example `EU`.
+            enum: [AS, AF, NA, SA, AN, EU, AZ]
+            example: EU
+          country:
+            description: Returns analytics based on the viewers' country. The list of supported country names are based on the [GeoNames public database](https://www.geonames.org/countries/). You must use the ISO-3166 alpha2 format, for example `FR`.
+            type: string
+            example: FR
+          deviceType:
+            type: string
+            description: 'Returns analytics based on the type of device used by the viewers. Possible response values are: `computer`, `phone`, `tablet`, `tv`, `console`, `wearable`, `unknown`.'
```
Comment (mioara-api-video) [inline openapi.yaml:15091]: Same comment here

File: openapi.yaml (hunk 1 above)
Comment (szekelyzol) [reply openapi.yaml:11946]: Added ✅

File: openapi.yaml (hunk 2 above)
Comment (szekelyzol) [reply openapi.yaml:12156]: Added ✅ @mioara-api-video `current-page` and `page-size` are just the reference links to the parameter schemas for pagination. The actual params are in `camelCase` ✅

File: openapi.yaml (hunk 4 above)
Comment (szekelyzol) [reply openapi.yaml:15060]: Good catch, you are right! Fixing ✅

File: openapi.yaml (hunk 3 above)
Comment (szekelyzol) [reply openapi.yaml:15052]: Added ✅

**Review by olivierapivideo — APPROVED** [review]

**Pull request merged**